source-academy/frontend
# Java slang

### Description



### Type of change


### How to test

The java compiler has not yet been integrated. To run a class file enter code in the editor in the format (we should remove this when the compiler is integrated):
```
{
"<classname>.class": "<base64 encoded compiled classfile>
}
```

main class is should be named `Main.class`

Compile with `javac *.java -target 8 -source 8 `

The standard library is missing almost all native method implementations so only simple Java programs can run.

Also note that lambda functions are quite slow.
![image](https://github.com/source-academy/frontend/assets/37036869/f0046c81-76bf-451c-94bb-395513d59b17)

### Checklist
